app/code/Magento/InventoryInStorePickup/Model/Carrier/Command/GetIsAnyPickupLocationAvailable.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Carrier\Command;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryInStorePickup\Model\GetPickupLocations;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Check if any pickup location is available in current scope.
+ */
+class GetIsAnyPickupLocationAvailable
+{
+    /**
+     * @var GetPickupLocations
+     */
+    private $getPickupLocations;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * Runtime cached values.
+     *
+     * @var array
+     */
+    private $isAvailable = [];
+
+    /**
+     * @param GetPickupLocations $getPickupLocations
+     * @param StockResolverInterface $stockResolver
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        GetPickupLocations $getPickupLocations,
+        StockResolverInterface $stockResolver,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->getPickupLocations = $getPickupLocations;
+        $this->stockResolver = $stockResolver;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * @param int $websiteId
+     *
+     * @return bool
+     */
+    public function execute(int $websiteId): bool
```

Comment:
It'd be better to rely on sales channel. Someone who puts a lot of effort into implementing any channel other then website would be grateful for your effort.

Reply:
Done.
Probably, it can be exposed as API endpoint.

Reply:
By the way, does it will be more logical for all of these services to use the same root:
```\Magento\InventoryInStorePickupApi\Api\GetNearbyPickupLocationsInterface
\Magento\InventoryInStorePickupApi\Api\GetPickupLocationsAssignedToStockOrderedByPriorityInterface
```